src/prefect/tasks/shell.py

```diff
@@ -123,8 +125,11 @@ def run(
                     if self.return_all:
                         lines.append(line)
 
-                    if self.stream_output:
-                        self.logger.debug(line)
+                    if isinstance(self.stream_output, bool):
+                        if self.stream_output:
+                            self.logger.info(line)
```

Comment:
I would instead do this determination at class initialization

```python
self.stream_output = logging.INFO if stream_output is True else stream_output
```

then here you can do

```python
if self.stream_ouput:
    self.logger.log(level=self.stream_output, msg=line)
```

Reply:
or if we want to support a user patching the stream output property live e.g

```python
my_task = ShellTask()
my_task.stream_output = False
```

then we'd want to use a property for `stream_output` and have that simple `if` logic exist in the setter

Reply:
It will return a string if not given an existing name so you could just check if it successfully cast to an int at the end

```
if self.stream_logs and not isinstance(self.stream_logs, int):
     raise TypeError(...)
```